Sources/Public/AnimationCache/LRUAnimationCache.swift

```diff
@@ -11,6 +11,8 @@ import Foundation
 ///
 /// Once `cacheSize` is reached, the least recently used animation will be ejected.
 /// The default size of the cache is 100.
+///
+/// Note: this cache doesn't evict entries under memory pressury, nor is thread-safe.
```

Comment:
I think it makes sense to deprecate it.

I thought about updating `LRUAnimationCache` like that PR did, but my concern is that the semantics would be a little off, as `NSCache` doesn't make any guarantees about which items would be evicted (or when).

Alternatively, we could:

1. Remove the existing `LRUAnimationCache`
1. Add a deprecate typealias `LRUAnimationCache` to the new implementation so it doesn't break any code, but raises awareness if anyone is using it directly and relying on its behavior

Reply:
I'm not too worried about the semantic change of adopting `NSCache`, since the semantics of the existing cache implementation isn't great or particularly useful. Lottie 4.0 is a fine time to make changes that affect the semantics of existing code.

So I'd personally prefer modifying the existing cache implementation rather than providing two. Renaming it and provided a deprecated typealias seems like a nice way to inform users about this change. If we do that I think `DefaultAnimationCache` may be a good name for it,

Reply:
Just deprecating `LRUAnimationCache` but leaving it as-is also seems fine. Whichever you prefer.

Reply:
Updated in 2b8aa68